spec/controllers/hub/faq_items_controller_spec.rb

```diff
@@ -0,0 +1,205 @@
+        expect(assigns(:position_options)).to eq [1, 2]
+        expect(response).to render_template :edit
+      end
+    end
+  end
+
+  describe "#update" do
+    let(:slug) { "how_moon" }
+    let(:params) do
+      {
+        faq_category_id: faq_category.id,
+        id: faq_item.id,
+        faq_item: {
+          faq_category_id: faq_category.id,
+          position: 2,
+          slug: slug,
+          question_en: "How will I go to the moon?",
+          question_es: "¿Cómo voy a ir a la luna?",
+          answer_en: "<div>In a tea cup</div>",
+          answer_es: "<div>En una taza de te</div>",
+        } }
+    end
+    it_behaves_like :an_action_for_admins_only, action: :update, method: :put
+
+    context "as an authenticated user" do
+      before do
+        sign_in user
+      end
+
+      it "updates the object based on passed params" do
+        put :update, params: params
+        faq_item.reload
+        expect(faq_item.position).to eq 2
+        expect(faq_item.question_en).to eq "How will I go to the moon?"
+        expect(faq_item.question_es).to eq "¿Cómo voy a ir a la luna?"
+        expect(faq_item.answer_en).to be_an_instance_of ActionText::RichText
+        expect(faq_item.answer_en.body.to_s).to include "<div>In a tea cup</div>"
+        expect(faq_item.answer_es).to be_an_instance_of ActionText::RichText
+        expect(faq_item.answer_es.body.to_s).to include "<div>En una taza de te</div>"
+        expect(faq_item.slug).to eq "how_moon"
+        expect(response).to render_template :show
+      end
+
+      it "shift other positions after it" do
+        put :update, params: params
+        faq_item.reload
+        expect(faq_item.position).to eq 2
+        expect(faq_item_2.reload.position).to eq 1
+      end
+
+      it "records a paper trail" do
+        put :update, params: params
+        faq_item.reload
+        expect(faq_item.versions.last.event).to eq "update"
+        expect(faq_item.versions.last.whodunnit).to eq user.id.to_s
+        expect(faq_item.versions.last.item_id).to eq faq_item.id
+        expect(faq_item.versions.last.item_type).to eq "FaqItem"
+      end
+
+      context "slug is empty" do
+        let!(:slug) { "" }
+        it "updates the slug to the parameterized version of the name" do
+          put :update, params: params
+          faq_item.reload
+          expect(faq_item.slug).to eq "how_will_i_go_to_the_moon"
+        end
+      end
+    end
+  end
+
+  describe "#new" do
+    context "as an authenticated user" do
+      before do
+        sign_in user
+      end
+
+      it "renders new" do
+        get :new, params: {faq_category_id: faq_category.id}
+        expect(assigns(:position_options)).to eq [1, 2, 3]
+        expect(response).to render_template :new
+      end
+    end
+  end
+
+  describe "#create" do
+    let(:params) do
+      {
+        faq_category_id: faq_category.id,
+        id: faq_item.id,
+        faq_item: {
+          faq_category_id: faq_category.id,
```

Comment:
Similar check to see if we need to resubmit this field (might be useful if we allow them to move items between categories, for example)